packages/core-data/src/entity-provider.js

```diff
@@ -0,0 +1,91 @@
+/**
+ * WordPress dependencies
+ */
+import { createContext, useContext, useCallback } from '@wordpress/element';
+import { useSelect, useDispatch } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { defaultEntities, kinds } from './entities';
+
+const entities = {
+	...defaultEntities.reduce( ( acc, entity ) => {
+		if ( ! acc[ entity.kind ] ) {
+			acc[ entity.kind ] = {};
+		}
+		acc[ entity.kind ][ entity.name ] = { context: createContext() };
+		return acc;
+	}, {} ),
+	...kinds.reduce( ( acc, kind ) => {
+		acc[ kind.name ] = new Proxy(
```

Comment:
I've been thinking here, it feels like there's unwanted complexity here (the whole entities object), it feels like what we need is just a global variable with two keys and where we generate a new context each time the value is inexistant (a map of map for instance)

Reply:
We still want it to not work if there is no entity config to avoid unnecessary confusion in digging for errors.

Here: https://github.com/WordPress/gutenberg/pull/17153/commits/c04c3f8cbde54480d941ba1f1e198a97aec935f9